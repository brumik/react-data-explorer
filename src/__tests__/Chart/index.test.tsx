import React from 'react'
import { render, waitFor } from '@testing-library/react';
import fetchMock from 'fetch-mock-jest';
import {
    stackedChartResponse,
    pieChartResponse
} from '../assets/fetchResponses';
import ChartRenderer from '../../components/Chart/index';
import {
    ChartKind,
    ChartTopLevelType,
    functions,
    ChartSchemaElement,
    ChartType
} from '../../index';

const charts: ChartSchemaElement[] = [
    {
        id: 1000,
        kind: ChartKind.wrapper,
        type: ChartTopLevelType.chart,
        parent: null,
        props: {
            height: 300
        },
        xAxis: {
            label: 'Date',
        },
        yAxis: {
            label: 'Jobs across all clusters'
        },
        api: {
            params: {},
            url: 'http://chart1.url'
        }
    },
    {
        id: 1001,
        kind: ChartKind.simple,
        type: ChartType.line,
        parent: 1000,
        props: {
            x: 'created_date',
            y: 'success_count'
        }
    },
    {
        id: 100,
        kind: ChartKind.wrapper,
        type: ChartTopLevelType.pie,
        parent: null,
        props: {
            height: 300,
            x: '',
            y: 'host_count'
        },
        api: {
            params: {
                group_by: 'org',
                include_others: true,
                attributes: ['host_count'],
                sort_by: `total_count:desc`
            },
            url: 'http://chart2.url',
            method: 'POST'
        }
    }
];


describe('Chart/Renderers/CreateWrapper', () => {
    afterEach(() => fetchMock.restore());

    xtest('should render all charts', async () => {
        fetchMock.post({ url: 'http://chart1.url' }, stackedChartResponse);
        fetchMock.post({ url: 'http://chart2.url' }, pieChartResponse);

        const { container } = render(<ChartRenderer
            data={{
                charts: charts,
                functions
            }}
        />);

        expect(container).toMatchSnapshot();
    });

    test('should render one chart specified in the id', async () => {
        fetchMock.post({ url: 'http://chart1.url' }, stackedChartResponse);
        fetchMock.post({ url: 'http://chart2.url' }, pieChartResponse);

        const { container } = render(<ChartRenderer
            ids={[1000]}
            data={{
                charts: charts,
                functions
            }}
        />);
        await waitFor(() => container.querySelector('svg'));

        expect(container).toMatchSnapshot();

    });

    test('should render two charts specified in the id', async () => {
        fetchMock.post({ url: 'http://chart1.url' }, stackedChartResponse);
        fetchMock.post({ url: 'http://chart2.url' }, pieChartResponse);

        const { container } = render(<ChartRenderer
            ids={[1000, 100]}
            data={{
                charts: charts,
                functions
            }}
        />);
        await waitFor(() => container.querySelector('svg'));

        expect(container).toMatchSnapshot();

    });
});