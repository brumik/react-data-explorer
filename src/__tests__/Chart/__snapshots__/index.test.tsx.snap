// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart/Renderers/CreateWrapper should render all charts 1`] = `
<div>
  <div>
    <div
      style="height: 300px;"
    />
  </div>
  <div>
    <div
      style="height: 300px;"
    />
  </div>
</div>
`;

exports[`Chart/Renderers/CreateWrapper should render one chart specified in the id 1`] = `
<div>
  <div>
    <div
      style="height: 300px;"
    >
      <div
        class="pf-c-chart"
        style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
      >
        <svg
          height="300"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 0 300"
          width="0"
        >
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="70"
              x2="-50"
              y1="230"
              y2="230"
            />
            <text
              direction="inherit"
              dx="0"
              id="chart-axis-0-axisLabel-0"
              x="10"
              y="281.97"
            >
              <tspan
                dx="0"
                dy="0"
                style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 40px; stroke: transparent; text-anchor: middle; fill: #252525;"
                text-anchor="middle"
                x="10"
              >
                Date
              </tspan>
            </text>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="-50"
                x2="-50"
                y1="230"
                y2="235"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-5"
                x="-50"
                y="256.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="middle"
                  x="-50"
                >
                  2020-11-05
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="70"
              x2="70"
              y1="50"
              y2="230"
            />
            <text
              direction="inherit"
              dx="0"
              id="chart-axis-1-axisLabel-0"
              transform="rotate(-90,30,140)"
              x="30"
              y="137.97"
            >
              <tspan
                dx="0"
                dy="0"
                style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 40px; stroke: transparent; text-anchor: middle; fill: #252525;"
                text-anchor="middle"
                x="30"
              >
                Jobs across all clusters
              </tspan>
            </text>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="183.24675324675323"
                y2="183.24675324675323"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-0"
                x="55"
                y="188.21675324675323"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  20
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="136.4935064935065"
                y2="136.4935064935065"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-1"
                x="55"
                y="141.4635064935065"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  40
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="89.74025974025975"
                y2="89.74025974025975"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-2"
                x="55"
                y="94.71025974025974"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  60
                </tspan>
              </text>
            </g>
          </g>
          <g
            clip-path="url(#victory-clip-2)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-2"
              >
                <rect
                  height="180"
                  vector-effect="non-scaling-stroke"
                  width="120"
                  x="-50"
                  y="50"
                />
              </clippath>
            </defs>
            <path
              d="M70,185.58441558441558L46,169.22077922077924L22,124.80519480519482L-2,185.58441558441558L-26.000000000000004,50L-50,230"
              role="presentation"
              shape-rendering="auto"
              style="fill: transparent; stroke: #06c; opacity: 1; stroke-width: 2;"
            />
          </g>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="300"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 0 300"
            width="0"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Chart/Renderers/CreateWrapper should render two charts specified in the id 1`] = `
<div>
  <div>
    <div
      style="height: 300px;"
    >
      <div
        class="pf-c-chart"
        style="pointer-events: none; position: relative; width: 100%; height: 100%;"
      >
        <svg
          height="300"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 0 300"
          width="0"
        >
          <g>
            <path
              d="M0,0Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #06c; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(10, 140)"
            />
            <path
              d="M0,0Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #8bc1f7; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(10, 140)"
            />
            <path
              d="M0,0Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #002f5d; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(10, 140)"
            />
            <path
              d="M0,0Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #519de9; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(10, 140)"
            />
            <path
              d="M0,0Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #004b95; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(10, 140)"
            />
            <path
              d="M0,0Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #06c; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(10, 140)"
            />
          </g>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="300"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 0 300"
            width="0"
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      style="height: 300px;"
    >
      <div
        class="pf-c-chart"
        style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
      >
        <svg
          height="300"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 0 300"
          width="0"
        >
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="70"
              x2="-50"
              y1="230"
              y2="230"
            />
            <text
              direction="inherit"
              dx="0"
              id="chart-axis-0-axisLabel-0"
              x="10"
              y="281.97"
            >
              <tspan
                dx="0"
                dy="0"
                style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 40px; stroke: transparent; text-anchor: middle; fill: #252525;"
                text-anchor="middle"
                x="10"
              >
                Date
              </tspan>
            </text>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="-50"
                x2="-50"
                y1="230"
                y2="235"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-5"
                x="-50"
                y="256.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="middle"
                  x="-50"
                >
                  2020-11-05
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="70"
              x2="70"
              y1="50"
              y2="230"
            />
            <text
              direction="inherit"
              dx="0"
              id="chart-axis-1-axisLabel-0"
              transform="rotate(-90,30,140)"
              x="30"
              y="137.97"
            >
              <tspan
                dx="0"
                dy="0"
                style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 40px; stroke: transparent; text-anchor: middle; fill: #252525;"
                text-anchor="middle"
                x="30"
              >
                Jobs across all clusters
              </tspan>
            </text>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="183.24675324675323"
                y2="183.24675324675323"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-0"
                x="55"
                y="188.21675324675323"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  20
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="136.4935064935065"
                y2="136.4935064935065"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-1"
                x="55"
                y="141.4635064935065"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  40
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                vector-effect="non-scaling-stroke"
                x1="70"
                x2="65"
                y1="89.74025974025975"
                y2="89.74025974025975"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-2"
                x="55"
                y="94.71025974025974"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                  text-anchor="end"
                  x="55"
                >
                  60
                </tspan>
              </text>
            </g>
          </g>
          <g
            clip-path="url(#victory-clip-5)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-5"
              >
                <rect
                  height="180"
                  vector-effect="non-scaling-stroke"
                  width="120"
                  x="-50"
                  y="50"
                />
              </clippath>
            </defs>
            <path
              d="M70,185.58441558441558L46,169.22077922077924L22,124.80519480519482L-2,185.58441558441558L-26.000000000000004,50L-50,230"
              role="presentation"
              shape-rendering="auto"
              style="fill: transparent; stroke: #06c; opacity: 1; stroke-width: 2;"
            />
          </g>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="300"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 0 300"
            width="0"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
